File: src/main/java/no/nav/syfo/rest/feil/Feilmelding.java
package no.nav.syfo.rest.feil;

import javax.ws.rs.core.Response.Status;

import static javax.ws.rs.core.Response.Status.*;

public class Feilmelding {

    static final String NO_BIGIP_5XX_REDIRECT = "X-Escape-5xx-Redirect";

    public enum Feil {
        ARBEIDSFORHOLD_UGYLDIG_INPUT (
                INTERNAL_SERVER_ERROR, "feilmelding.arbeidstaker.ugyldig.input"
        ),
        ARBEIDSFORHOLD_INGEN_TILGANG (
                INTERNAL_SERVER_ERROR, "feilmelding.arbeidstaker.sikkerhetsbegrensning"
        ),
        ARBEIDSFORHOLD_GENERELL_FEIL (
                INTERNAL_SERVER_ERROR, "feilmelding.arbeidstaker.generell.feil"
        ),
        GENERELL_FEIL (
                INTERNAL_SERVER_ERROR, "feilmelding.generell.feil"
        ),
        IKKE_FOEDSELSNUMMER (
                INTERNAL_SERVER_ERROR, "feilmelding.ikke.fnr"
        ),
        INGEN_AKTOER_ID (
                INTERNAL_SERVER_ERROR, "feilmelding.ingen.aktoer.id"
        ),
        AKTOER_IKKE_FUNNET (
                INTERNAL_SERVER_ERROR, "feilmelding.aktoer.ikke.funnet"
        );

        Status status;
        String id;
        Feil(Status status, String id) {
            this.status = status;
            this.id = id;
        }
    }

    private Feil feil;

    Feilmelding withFeil(final Feil feil) {
        this.feil = feil;
        return this;
    }

    @SuppressWarnings("unused")
    public String getId() {
        return feil.id;
    }
}

File: src/main/kotlin/no/nav/syfo/api/exception/Feilmelding.kt
package no.nav.syfo.api.exception

import javax.ws.rs.core.Response

class Feilmelding {
    enum class Feil(var status: Response.Status, var id: String) {
        ARBEIDSFORHOLD_UGYLDIG_INPUT(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.arbeidstaker.ugyldig.input"
        ),
        ARBEIDSFORHOLD_INGEN_TILGANG(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.arbeidstaker.sikkerhetsbegrensning"
        ),
        ARBEIDSFORHOLD_GENERELL_FEIL(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.arbeidstaker.generell.feil"
        ),
        GENERELL_FEIL(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.generell.feil"
        ),
        IKKE_FOEDSELSNUMMER(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.ikke.fnr"
        ),
        INGEN_AKTOER_ID(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.ingen.aktoer.id"
        ),
        AKTOER_IKKE_FUNNET(Response.Status.INTERNAL_SERVER_ERROR, "feilmelding.aktoer.ikke.funnet"
        );
    }

    private var feil: Feil? = null
    fun withFeil(feil: Feil?): Feilmelding {
        this.feil = feil
        return this
    }

    val id: String
        get() = feil!!.id

    companion object {
        const val NO_BIGIP_5XX_REDIRECT = "X-Escape-5xx-Redirect"
    }
}
